docs/tools/terraform/reference/cookbook/clickhouse-access-setup-recipe.rst

```diff
@@ -0,0 +1,245 @@
+correct permissions for two roles: the writer role (allowed to insert data) and the analyst role (allowed to query data).
+
+Configure common files
+''''''''''''''''''''''
+
+.. dropdown:: Expand to check out the relevant common files needed for this recipe.
+
+  Navigate to a new folder and add the following files:
+
+  1. ``provider.tf`` file
+
+    .. code:: terraform
+
+       terraform {
+	 required_providers {
+	   aiven = {
+	     source  = "aiven/aiven"
+	     version = ">= 3.8"
+	   }
+	 }
+       }
+
+       provider "aiven" {
+	 api_token = var.aiven_api_token
+       }
+
+  .. tip::
+    
+    You can set environment variable ``AIVEN_TOKEN`` for the ``api_token`` property so that you don't need to pass the ``-var-file`` flag when executing Terraform commands.
+
+  2. ``variables.tf`` file
+
+  Use it for defining the variables to avoid including sensitive information in source control. The ``variables.tf`` file defines the API token, the project name to use, and the prefix for the service name:
+
+    .. code:: terraform
+
+       variable "aiven_api_token" {
+	 description = "Aiven console API token"
+	 type        = string
+       }
+
+       variable "project_name" {
+	 description = "Aiven console project name"
+	 type        = string
+       }
+
+  3. ``*.tfvars`` file
+
+  Use it to indicate the actual values of variables so that they can be passed (with the ``-var-file=`` flag) to Terraform during runtime and excluded later on. Configure the ``var-values.tfvars`` file as follows:
+
+    .. code:: terraform
+
+       aiven_api_token     = "<YOUR-AIVEN-AUTHENTICATION-TOKEN-GOES-HERE>"
+       project_name        = "<YOUR-AIVEN-CONSOLE-PROJECT-NAME-GOES-HERE>"
+
+Set up the service with the database
+''''''''''''''''''''''''''''''''''''
+
+Configure the ``services.tf`` file as follows:
+
+.. code:: terraform
+
+  resource "aiven_clickhouse" "clickhouse" {
+    project                 = var.project_name
+    cloud_name              = "google-europe-west1"
+    plan                    = "startup-16"
+    service_name            = "clickhouse-gcp-eu"
+    maintenance_window_dow  = "monday"
+    maintenance_window_time = "10:00:00"
+  }
+
+  resource "aiven_clickhouse_database" "measurements" {
+    project                 = var.project_name
+    service_name            = aiven_clickhouse.clickhouse.service_name
+    name                    = "iot_measurements"
+  }
+
+.. topic:: Expected result
+
+  * ``"aiven_clickhouse"`` resource creates an Aiven for ClickHouse service with the project name, the cloud name (provider, region, zone), the service plan, and the service name as specified in the ``services.tf`` file.
+  * ``"aiven_clickhouse_database"`` resource creates a database with the project name, the service name, and the database name as specified in the ``services.tf`` file.
+
+Grant user's permissions
+''''''''''''''''''''''''
+
+Administrator role - read & write access
```

Comment:
It's not really an administrator role since the writer cannot for instance delete (drop) databases if I understand correctly. Maybe writer would be more appropriate, altough a bit repetitive I agree.

Reply:
@

> It's not really an administrator role since the writer cannot for instance delete (drop) databases if I understand correctly. Maybe writer would be more appropriate, altough a bit repetitive I agree.

@aris-aiven , how about `a contributor role`, i.e. someone who can add but not delete?

Reply:
But I guess you don't contribute to a database? Manager, writer, publisher all seem fine, and I'm open to any other suggestion as well.

Reply:
> But I guess you don't contribute to a database? Manager, writer, publisher all seem fine, and I'm open to any other suggestion as well.

ok, I'll switch to `writer` if you find it more appropriate than `contributor`. Thanks!